Source/Protocols/Focusable.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  Focusable.swift
+//  Matrioska
+//
+//  Created by Andreas Thenn on 14/03/2017.
+//  Copyright © 2017 runtastic. All rights reserved.
+//
+
+import Foundation
+
+/** This protocol should be implemented by every UIViewController that has some kind of focus,
+ highlight or select method. We want these to be adressable in a more generic way and
+ we don't need to care about the actual underlying method that's being called.
+ */
+protocol Focusable {
+
+    /** Try to focus on the given UIViewController, otherwise it fails silently */
+    @discardableResult
+    func focus(on viewController: UIViewController)
```

Comment:
I second @joanromano's position: _"I would expect Matrioska to support navigation to some Component or child Component in a generic way"_. Matrioska is basically a "`Component` tree API", so the API should allow clients to focus `Component`s, not `UIViewController`s. 

But before that, I think we need to introduce an identity concept for `Component`s, a means by which `Component`s can be looked up, so that you could find the `Component` to focus.

The question is also whether the focus request should be dispatched to the root component, i.e.
```
root.focus(componentId: someId)
```
or the `Component` to focus: 
```
let someNestedComponent = root.find(by: someId)
root.focus(component: someNestedComponent)
```